src/utilities/inVariable.js

```diff
@@ -1,14 +1,10 @@
-import _ from 'lodash';
+const inVariable = (value) => {
+  // check if the user has selected more than just a variable
+  if (value.selection.start.key !== value.selection.end.key) return false;
 
-const inVariable = (descendants) => {
-  if (descendants.size !== 4) return false;
-  const pattern = descendants.map(node => node.type).toArray();
-  if (
-    // pattern of nodes when in a variable is the following:
-    // clause, paragraph, variable, undefined (text node has no type)
-    _.isEqual(pattern, ['clause', 'paragraph', 'variable', undefined])
-    && descendants.get(descendants.size - 1).object === 'text' // check that node with undefined type is text node
-  ) return true;
+  const descendants = value.document.getDescendantsAtRange(value.selection);
+  // check that the selection contains exactly one variable node
+  if (descendants.filter(d => d.type === 'variable').size === 1) return true;
```

Comment:
Am I missing something, but wouldn't that still allow edits if the selection starts inside fixed text and end inside a variable?

Reply:
Line 3 before this is what prevents that. If the start and end points have two different keys, we immediately return false (so what we looked at on our call worked!).